include/geos/geom/LineSegment.h

```diff
@@ -250,13 +250,20 @@ class GEOS_DLL LineSegment {
     //
     /// @param ret will be set to the midpoint of the segment
     ///
-    void midPoint(Coordinate& ret) const
+    Coordinate midPoint() const
```

Comment:
Suggest using `CoordinateXY` anywhere we are not accessing `z`.

Reply:
Good point.